adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/utils/GeodeticUtils.java

```diff
@@ -0,0 +1,106 @@
+import org.locationtech.spatial4j.shape.ShapeFactory;
+
+import static org.locationtech.spatial4j.context.SpatialContext.GEO;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.ShapePoint;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceCalculationUtils;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceUnit;
+
+import java.util.Map;
+
+/**
+ * Interface implementation using external library org.locationtech.spatial4j. This is used to compute distance between
+ * two {@code ShapePoint}.
+ */
+public class GeodeticUtils implements DistanceCalculationUtils {
+
+    /**
+     * Return the distance between two points given there lat/lon positions. Note that points of origin and destination
+     * can be swapped.
+     *
+     * @param originLatitude        latitude of the origin point
+     * @param destinationLatitude   latitude of the destination point
+     * @param originLongitude       longitude of the origin point
+     * @param destinationLongitude  longitude of the destination point
+     * @param distanceUnit          unit of the desired result of computation
+     * @return the distance between two points given there lat/lon positions in the specified unit
+     */
+    @Override
+    public double distanceBetweenTwoPoints(final float originLatitude, final float destinationLatitude,
+                                                  final float originLongitude, final float destinationLongitude,
+                                                  final DistanceUnit distanceUnit) {
+        final ShapeFactory shapeFactory= getShapeFactory();
+        final DistanceCalculator distanceCalculator = getDistanceCalculator();
+
+        final Point origin = shapeFactory.pointXY(originLongitude, originLatitude);
+        final Point destination = shapeFactory.pointXY(destinationLongitude, destinationLatitude);
+        final double distance = distanceCalculator.distance(origin, destination);
+
+        if (distanceUnit == DistanceUnit.METER) {
+                return distance;
+        } else {
+                return distance/KILOMETER_TO_METER_CONVERSION_FACTOR;
+        }
+    }
+
+    /**
+     * private method returning a {@code ShapeFactory}
+     * @return a {@link ShapeFactory}
+     */
+    private static ShapeFactory getShapeFactory() {
+        return GEO.getShapeFactory();
+    }
+
+    /**
+     * private method returning a {@code DistanceCalculator}
+     * @return a {@link DistanceCalculator}
+     */
+    private static DistanceCalculator getDistanceCalculator() {
+        return GEO.getDistCalc();
+    }
+
+    /**
+     * Method returns the total length of a shape. As a reminder, a shape is a collection of {@code ShapePoint} ordered
+     * by shape_pt_sequence. The result is expressed in the specified unit.
+     *
+     * @param shape          the collection of {@link ShapePoint} whose total length is to be computed
```

Comment:
```suggestion
     * @param shape          the collection of {@link ShapePoint} sorted by `shape_dist_traveled` whose total length is to be computed
```

Reply:
I get the idea, but aren't `ShapePoint` sorted by `shape_pt_sequence`?

Reply:
Yes, but it doesn't hurt to clarify on this method that the input needs to sorted to get a correct result. I typically try to document assumptions in input when it affects correctness of the output, even if they are documented elsewhere.